pkg/agent/querier/querier.go

```diff
@@ -193,12 +193,15 @@ func (aq agentQuerier) GetAgentInfo(agentInfo *v1beta1.AntreaAgentInfo, partial
 		agentInfo.Version = querier.GetVersion()
 		agentInfo.PodRef = querier.GetSelfPod()
 		agentInfo.NodeRef = querier.GetSelfNode(true, aq.nodeConfig.Name)
+		// Make a new string slice instead of appending agentInfo.NodeSubnet directly to avoid duplicate CIDRs.
```

Comment:
Why does it cause duplicate CIDRs?

Reply:
In the case of restarting Agent, according to current monitoring handling workflow, we will update the entire CRD (https://github.com/vmware-tanzu/antrea/blob/master/pkg/monitor/agent.go#L79). But monitor.agentCRD is not nil and is likely to have nodeSubnet (https://github.com/vmware-tanzu/antrea/blob/master/pkg/monitor/agent.go#L62), to avoid duplicate subnet, we need to initialize a new slice for nodeSubnet every time we need to update the entire CRD.

Reply:
> In the case of restarting Agent, according to current monitoring handling workflow, we will update the entire CRD (https://github.com/vmware-tanzu/antrea/blob/master/pkg/monitor/agent.go#L79). But monitor.agentCRD is not nil and is likely to have nodeSubnet (https://github.com/vmware-tanzu/antrea/blob/master/pkg/monitor/agent.go#L62), to avoid duplicate subnet, we need to initialize a new slice for nodeSubnet every time we need to update the entire CRD.

Got it, thanks for clarifying.